image-processing/main.py

```diff
@@ -20,10 +23,43 @@ def trigger_event(event, context):
         None; the output is written to Stackdriver Logging
     """
 
-    print("Event ID: {}".format(context.event_id))
-    print("Event type: {}".format(context.event_type))
-    print("Bucket: {}".format(event["bucket"]))
-    print("File: {}".format(event["name"]))
-    print("Metageneration: {}".format(event["metageneration"]))
-    print("Created: {}".format(event["timeCreated"]))
-    print("Updated: {}".format(event["updated"]))
+    bucket_name = event["bucket"]
+    source_img = event["name"]
+    tmp_download_path = f"/tmp/{source_img}"
+
+    try:
+        download_image(bucket_name, source_img, tmp_download_path)
+        print(f"Image {source_img} downloaded to {tmp_download_path}")
+        # TODO(abi) call cloud vision API with this image
+        # TODO(jerry) call R script with features returned from Abi's part
+        # TODO(hantao) put processed images to `sie-processed-images` bucket
+    except Exception:
+        pass
```

Comment:
I am not sure how Python handles exceptions but is this an appropriate way to do it?

Reply:
it is not but I just left it there for you guys to add the specific exceptions raised by your methods